### DISCUSSION TITLE
DatasetSummary Min and Max on image foreground

### DISCUSSION THREAD

**Discussion by NabJa** [Project-MONAI/MONAI#3819] [General]
Hello MONAI community,

In the [DatasetSummary ](https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/dataset_summary.py#L28) class the min and max voxel intensities are computed on the whole image ([here](https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/dataset_summary.py#L143)).  
However, in the nnUNet implementation of the DatasetAnalyzer does extract min and max voxel intensities ([here](https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunet/experiment_planning/DatasetAnalyzer.py#L195)) only on the foreground (see [foreground extraction](https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunet/experiment_planning/DatasetAnalyzer.py#L188)).

Shouldn't the min and max extraction of the voxels be moved to after the foreground extraction?

Thanks in advance for your opinions!
Nabil

**Comment by wyli** [comment]
thanks for the pointer, let me create a ticket on this...

**Reply by Nic-Ma** [reply]
Hi @dongyang0122 , what do you think about this question? May I know the thoughts when you develop the `DatasetSummary`?

Thanks in advance.

**Reply by dongyang0122** [reply]
@Nic-Ma sorry I have no clue about `DatasetSummary`

**Comment by dongyang0122** [comment]
Yes, we need intensity stats on foreground regions. But we need more than that. For example, a common practice is using 5% and 95% of overall intensity distribution as thresholds for intensity clipping.

**Reply by yiheng-wang-nv** [reply]
Hi @dongyang0122 , for percentiles, can you use `calculate_percentiles`?  See: https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/dataset_summary.py#L155

**Comment by Nic-Ma** [comment]
Sorry I misunderstood this module, it was developed by @yiheng-wang-nv not @dongyang0122 's AutoML module.

Hi @yiheng-wang-nv , could you please help share some comments? Or maybe you can enhance the features?

Thanks in advance.

**Reply by yiheng-wang-nv** [reply]
Hi @NabJa , thanks! Let me submit a PR to modify it.
